internal/cluster/distribution/pke/service.go

```diff
@@ -120,12 +124,18 @@ type service struct {
 	genericClusters Store
 	clusterManager  ClusterManager
 	nodePoolManager NodePoolManager
+	nodePoolStore   NodePoolStore
 }
 
 // +testify:mock:testOnly=true
 
 // NodePoolManager is responsible for managing node pools.
 type NodePoolManager interface {
+	// DeleteNodePool deletes an existing node pool in a cluster.
+	DeleteNodePool(
+		ctx context.Context, c cluster.Cluster, existingNodePool ExistingNodePool, shouldUpdateClusterStatus bool,
```

Comment:
What kind of formatting is this? If the line is too long, break every parameter into a new line.

Reply:
I believe this is on me.

I have been experimenting with a 3 step function signature formatting approach in some of my previous PRs where it was approved so I highlighted this possibility of formatting for Jani.

(In order of precedence.)

1. **Single line formatting**: If the function signature fits into 120 characters, then it should be formatted to a single line.
```golang
func (receiver receiverType) functionName(parameter1 parameterType1, parameter2 parameterType2) (returnValue1 returnType1, returnValue2 returnType2)
```

2. **Single line parameter list**: If the function signature doesn't fit into 120 characters, but the function parameter list fits into 120 characters, then the function name, the parameter list and the return values should be formatted into their own dedicated lines.
```golang
func (receiver receiverType) functionName(
    parameter1 parameterType1, parameter2 parameterType2, parameter3 parameterType3, parameter4 parameterType4, parameter5 parameterType5,
) (returnValue1 returnType1, returnValue2 returnType2)
```

3. **One parameter per line**: If the function parameter list doesn't fit into 120 characters, then the function name, each individual parameter and the return values should be formatted into their own dedicated lines.
```golang
func (receiver receiverType) functionName(
    parameter1 parameterType1,
    parameter2 parameterType2, 
    parameter3 parameterType3, 
    parameter4 parameterType4, 
    parameter5 parameterType5,
) (returnValue1 returnType1, returnValue2 returnType2)
```

The reason why I introduced the single-line-parameter-list formatting is because the one-parameter-per-line formatting occasionally in minimally arrow-head like situations with long function names takes up so much vertical space with very short arguments, it makes the code more difficult to read IMO, although this is debatable.

Examples:
```golang
// Original.
func Name(c cluster.Cluster) {
    // ...
    if condition {
        // ...
        err := operation()
        if err != nil {
            err = methodWithAVeryLongQualifiedName(many, arguments, with, short, names, barely, over, the, line, length, limits)

            return err
        }
        // ...
    }
    // ....
}
// single-line-parameter-list.
func Name(c cluster.Cluster) {
    // ...
    if condition {
        // ...
        err := operation()
        if err != nil {
            err = methodWithAVeryLongQualifiedName(
                many, arguments, with, short, names, barely, over, the, line, length, limits
            )

            return err
        }
        // ...
    }
    // ...
}
// one-parameter-per-line.
func Name(c cluster.Cluster) {
    // ...
    if condition {
        // ...
        err := operation()
        if err != nil {
            err = methodWithAVeryLongQualifiedName(
                many, 
                arguments, 
                with, 
                short, 
                names, 
                barely, 
                over, 
                the, 
                line, 
                length, 
                limits,
            )

            return err
        }
        // ...
    }
    // ...
}
```

I apologize for the confusion, I should have highlighted the single-line-parameter-list formatting in the PR's I experimented with to make sure it is not overlooked.

I'm fine with dropping the single-line-parameter-list formatting and reverting back to the single-line or one-parameter-per-line 2 step function signature formatting approach.
Although I still feel like in some edge cases the single-line-parameter-list formatting approach is preferable, I can accept it being opinionated and thus not favorable.

Reply:
I can correct these in a future PR, it's not difficult to find them by regular expression.